gitspy/Repository.swift

```diff
@@ -17,4 +17,55 @@ struct Repository {
     let newStars: Int
     let totalStars: Int
     
+    // MARK: Network
+
+    static func repos(userId: String, completion: (parsedRepos: [Repository]?) -> Void) {
+        let userResource = Resource(pathComponent: "repos?user_id=" + userId) { (object) -> [Repository]? in
+            guard let dictionary = object as? [String: AnyObject],
+                let repos = dictionary["repos"] as? [[String: AnyObject]] else {
+                return nil
+            }
+            
+            return Repository.array(repos)
+        }
+        
+        Networker.sharedInstance.loadResource(userResource, callback: completion)
+    }
+    
+    static func watch(repoId: Int, completion: (status: Bool?) -> Void) {
+        let boolResource = Resource(pathComponent: "repos/" + String(repoId), method: "POST") { (object) -> Bool? in
+            guard let status = object["status"] as? String else {
+                return nil
+            }
+
+            return status == "ok"
+        }
+        
+        Networker.sharedInstance.loadResource(boolResource, callback: completion)
+    }
```

Comment:
Don't you think the models would do too many things in this way? Now they are in both parsing (which is probably fine) and throwing network stuff. What if we add more endpoints to the Repository model? I think it would tend to be really huge this way.